bundles/AdminBundle/Controller/Admin/Asset/AssetController.php

```diff
@@ -217,6 +218,24 @@ static function (Task $task) {
             $asset->getScheduledTasks()
         );
 
+        $userOwner = User::getById($asset->getUserOwner());
+        if (empty($userOwner)) {
+            $data['userOwnerUsername'] = '';
+            $data['userOwnerFullname'] = $this->trans('user_unknown');
+        } else {
+            $data['userOwnerUsername'] = $userOwner->getName();
+            $data['userOwnerFullname'] = $userOwner->getFullName();
+        }
+
+        $userModification = ($asset->getUserOwner() == $asset->getUserModification()) ? $userOwner : User::getById($asset->getUserModification());
+        if (empty($userModification)) {
+            $data['userModificationUsername'] = '';
+            $data['userModificationFullname'] = $this->trans('user_unknown');
+        } else {
+            $data['userModificationUsername'] = $userModification->getName();
+            $data['userModificationFullname'] = $userModification->getFullName();
+        }
+
```

Comment:
We have similar code (for username and usermodification) in DataObjectController, DocumentController etc. 
If we add this code in a trait, then different controllers can access it from there.

Reply:
The new trait has been added, please review the updated code.

I have also made a fix to the metainfo panel so it handles the system account properly.

Reply:
Thank you! . Could you please resolve the conflicts also ?